src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -575,35 +575,34 @@ public static int GetReplicaCount(this IResource resource)
     /// Gets the deployment target for the specified resource, if any. Throws an exception if
     /// there are multiple compute environments and a compute environment is not explicitly specified.
     /// </summary>
-    public static DeploymentTargetAnnotation? GetDeploymentTargetAnnotation(this IResource resource)
-    {
 #pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+    public static DeploymentTargetAnnotation? GetDeploymentTargetAnnotation(this IResource resource, IComputeEnvironmentResource? computeEnvironmentResource = null)
+    {
         if (resource.TryGetLastAnnotation<ComputeEnvironmentAnnotation>(out var computeEnvironmentAnnotation))
         {
-            // find the deployment target for the compute environment
-            return resource.Annotations
-                .OfType<DeploymentTargetAnnotation>()
-                .LastOrDefault(a => a.ComputeEnvironment == computeEnvironmentAnnotation.ComputeEnvironment);
+            // If the resource has a ComputeEnvironmentAnnotation, use it to get the compute environment.
+            // This wins over the specified computeEnvironmentResource
```

Comment:
This seems like odd behavior. I explicitly said "environmentFoo", but the method return "environmentBar".

Reply:
Now the environmentResource is a filter.